WDSUS-44 PHP 8.0 dependency updates

asharirfan opened the pull request:
# Closes
WDSUS-44

<!--
Please do not leave this blank, use this format to link issue numbers:
[#WDS-123](https://webdevstudios.atlassian.net/browse/TICKET-NUMBER)
-->

## What type of PR is this? (put an x to all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Link to test

<!--
Please do not leave this blank, use this format to link to testing environment:
[Develop](https://wds.wdslab.com/)
-->

## Description

This PR:

- Updates composer and node dependencies
- Fix linting issues
- Tested the theme with PHP 8.2

asharirfan committed:
WDSUS-44 Update composer dependency packages

asharirfan committed:
WDSUS-44 Remove unused parameter from excerpt-more function

asharirfan committed:
WDSUS-44 Remove extra line

asharirfan committed:
WDSUS-44 Move global constants from blocks scaffold class to functions.php

asharirfan committed:
WDSUS-44 Move wpcli functions from class to separate files

asharirfan committed:
WDSUS-44 Fix phpcs properties

asharirfan committed:
WDSUS-44 Fix linting issues

asharirfan committed:
WDSUS-44 Update package json dependencies

asharirfan committed:
WDSUS-44 Fix cta pattern slug

asharirfan committed:
WDSUS-44 Rename block pattern categories registeration functions

asharirfan committed:
Merge branch 'release/WDSUS-2-4.0-Theme-Revamp' into feature/WDSUS-44-dependency-updates

khleomix reviewed (changes requested):
@asharirfan Please fix conflicts. Thanks!!!

asharirfan committed:
Merge branch 'main' into feature/WDSUS-44-dependency-updates

asharirfan committed:
WDSUS-44 Downgrade wordpress/eslint-plugin from 16.0 to 15.1

Downgrading the package because there is a dependency conflict. `npm run lint:js` was returning error that eslint-plugin-prettier package is missing.

asharirfan commented:
@khleomix This is ready for your review.

khleomix committed:
Merge branch 'main' into feature/WDSUS-44-dependency-updates

khleomix committed:
WDSUS-44 Update eslint-plugin

khleomix committed:
WDSUS-44 - Update postcss

khleomix committed:
WDSUS-44 - Update composer

khleomix committed:
WDSUS-44 - Update package lock

khleomix committed:
WDSUS-44 - Update php tested version

khleomix committed:
WDSUS-44 - Update eslint and prettier

khleomix committed:
WDSUS-44 - Update assertions and codeql workflows

khleomix committed:
WDSUS-44 - Update changelog and lefthook to ignore changelog

khleomix reviewed (approved):
Thanks @asharirfan I've added additional changes/fixes but nice work!
